### PR TITLE
streamlit 1.40.0 

### PR DESCRIPTION
**Pull request by Cansisti**
streamlit 1.40.0 

**Destination channel:** Defaults

### Links

- [PKG-6153]
- dev_url:        https://github.com/streamlit/streamlit/tree/1.40.0
- conda_forge:    https://github.com/conda-forge/streamlit-feedstock
- pypi:           https://pypi.org/project/streamlit/1.40.0
- pypi inspector: https://inspector.pypi.io/project/streamlit/1.40.0

### Explanation of changes:

- new version number

[PKG-6153]: https://anaconda.atlassian.net/browse/PKG-6153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {% set name = "streamlit" %}
-{% set version = "1.38.0" %}
+{% set version = "1.40.0" %}
```
Comment (cbouss) [inline recipe/meta.yaml:2]: Can you update to 1.40.1 instead? It was released two weeks ago.

Comment (lorepirri) [reply recipe/meta.yaml:2]: We have a SF request for this version, in case we can bump it after releasing `1.40.0`.

Comment (tferguson47) [reply recipe/meta.yaml:2]: Yeah this is a specific versioned request.  SF builds these internally and wants us to match versions.  if it is low effort to do I am okay with Tomasz building the newer version as well

CC @cbouss @lorepirri @Cansisti

Comment (Cansisti) [reply recipe/meta.yaml:2]: I can build 1.40.1 in another PR if you want it @cbouss

**Review by cbouss — APPROVED** [review]

**Pull request merged**